core/src/trezor/ui/layouts/mercury/__init__.py

```diff
@@ -970,48 +976,26 @@ async def confirm_ethereum_tx(
         recipient: str,
         total_amount: str,
         maximum_fee: str,
-        items: Iterable[tuple[str, str]],
+        fee_info_items: Iterable[tuple[str, str]],
         br_name: str = "confirm_ethereum_tx",
         br_code: ButtonRequestType = ButtonRequestType.SignTx,
         chunkify: bool = False,
     ) -> None:
-        info_layout = RustLayout(
-            trezorui2.show_info_with_cancel(
-                title=TR.confirm_total__title_fee,
-                items=items,
-            )
+        await confirm_output(
+            recipient,
+            title=TR.words__recipient,
+            chunkify=chunkify,
+            cancel_text=TR.buttons__cancel,
         )
 
-        while True:
-            # Allowing going back and forth between recipient and summary/details
-            await confirm_blob(
-                br_name,
-                TR.words__recipient,
-                recipient,
-                verb=TR.buttons__continue,
-                chunkify=chunkify,
-                prompt_screen=False,
-            )
-
-            try:
-                total_layout = RustLayout(
-                    trezorui2.confirm_total(
-                        title=TR.words__title_summary,
-                        items=[
-                            (f"{TR.words__amount}:", total_amount),
-                            (TR.send__maximum_fee, maximum_fee),
-                        ],
-                        info_button=True,
-                        cancel_arrow=True,
-                    )
-                )
-                total_layout.request_complete_repaint()
-                await raise_if_not_confirmed(
-                    with_info(total_layout, info_layout, br_name, br_code)
-                )
-                break
-            except ActionCancelled:
-                continue
+        await _confirm_summary(
+            items=(
+                (TR.words__amount, total_amount),
+                (TR.send__maximum_fee, maximum_fee),
+            ),
+            fee_items=fee_info_items,
+            cancel_text=TR.buttons__cancel,
+        )
```

Comment:
This is a regression from previous version because we cannot go back and forth between recipient and summary, no?

Reply:
According to the design I don't see a way to go back. In the old design you would go to the Cancel screen in the 2nd step and that would let you go back to the 1st step. Now the Cancel screen is called *Cancel sign*, so it should cancel the whole flow rather than just the 2nd step. TBH, the old way was a bit clunky IMO. I do expect the Cancel screen to cancel everything, not go back one step. What we could do, maybe, is use swipe down to go back.

Reply:
You go back to the recipient from summary screen by swiping down.

And yes, cancel should cancel the whole flow. On TT, there is an upper arrow. See [here](https://www.figma.com/design/l0gG9XeRJ8FTDQ3cfb1wv9/Firmware?node-id=8833-74263&t=TV8t9vF9ak72q97n-4).

Reply:
Still, it is a regression from model T and model R where you can browse through recipient and summary. 
Let's take model T as an example, the 2nd part (summary) has the close button `x` replaced with arrow up `^`, which is a workaround so that it seems like you go back, in fact you cancel the 2nd screen and catch `ActionCancelled`. This approach does not translate into mercury, where you have "Cancel" button. 

Possible solutions:
1. wrap it again into an infinite loop in uPy and figure out a control logic which breaks it if you cancel any window while also enabling going back and forth.
This might be difficult because we need to distinguish somehow what the user wants from the return object sent to uPy, we cannot distinguish whether `CANCELLED` comes from swipe down action of the summary or clicking the "Cancel" button. Maybe we can misuse `INFO` but I wouldn't go this way. Introducing a new object seems like an overkill but perhaps it might make sense... something like `GO_PREVIOUS` :thinking: 
2. pack all the arguments for rust and call just one flow layout. That would mean merging `confirm_output` and `confirm_summary` into something like `flow_confirm_altcoin_tx` - this might take longer to program but is the correct solution IMHO. We could handle swiping here and there.

Reply:
https://github.com/trezor/trezor-firmware/pull/4095/commits/acb1cdeadb4ef891c6bdcfb3e0220ca49a9ed4f6 Here is the solution.

Reply:
Swipe up works now. Also, opening the menu then closing the menu brings you back to the correct screen which you were on before the menu got opened.

Clicking *Cancel sign* then canceling the *Cancel* operation however brings you back to the beginning of the flow, as opposed to the screen you were on when you entered the menu that got you to *Cancel sign*. It's a pretty trivial fix to do it the other way, if @Hannsek thinks it is better UX. I am not sure honestly which way is better... I am guessing that if you were about to cancel, then decided to not cancel... you might as well review all steps again.

BTW, this is now one flow (*confirm_output*), with optional steps for *amount* and *summary* (which in turn goes further to *sign*). It should probably be reused in other cases as well so that we can simplify some of the other use cases. I'll have a look where else it would be a fit.

Reply:
@obrusvit generally, yes
However, the behavior of flows is not programmed at one place. You have a way to program the transitions for each flow separately, so a careful review is needed.